.github/workflows/prover.yml

```diff
@@ -0,0 +1,41 @@
+name: Run prover guest code checks
+
+on:
+  pull_request:
+  merge_group:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          toolchain: nightly-2024-07-27
+      
+      # Installs mold (modern ld), a drop-in replacement for lld.
+      # Under the hood, the following actions symlinks mold binary onto lld,
+      # so everything is linked faster (hopefully).
+      - name: Install mold linker
+        uses: rui314/setup-mold@v1
```

Comment:
`mold` is awesome. Great addition.

Reply:
Agree. 

Honestly, I didn't know much about other linkers, but just recently discovered it as part of this task (while digging various articles about how people **actually** **optimize compile time**). It shows super promising results!

Reply:
Too bad that it's abandone for MacOS